src/snowflake/snowpark/mock/_functions.py

```diff
@@ -495,45 +501,63 @@ def mock_to_time(
             if data is None:
                 res.append(None)
                 continue
-            if auto_detect and data.isnumeric():
-                res.append(
-                    datetime.datetime.utcfromtimestamp(
-                        process_numeric_time(data)
-                    ).time()
-                )
-            else:
-                # handle seconds fraction
-                data_parts = data.split(".")
-                if len(data_parts) == 2:
-                    # there is a part of seconds
-                    seconds_part = data_parts[1]
-                    # find the idx that the seconds part ends
-                    idx = 0
-                    while seconds_part[idx].isdigit():
-                        idx += 1
-                    # truncate to precision
-                    seconds_part = (
-                        seconds_part[: min(idx, fractional_seconds)]
-                        + seconds_part[idx:]
-                    )
-                    data = f"{data_parts[0]}.{seconds_part}"
-                res.append(
-                    (
-                        datetime.datetime.strptime(
-                            process_string_time_with_fractional_seconds(
-                                data, fractional_seconds
-                            ),
-                            time_fmt,
-                        )
-                        + datetime.timedelta(hours=hour_delta)
-                    ).time()
+            if auto_detect:
+                # handling non-standard string expr cases
+                if isinstance(column.sf_type.datatype, StringType) and data.isnumeric():
```

Comment:
NIT: Use `isdigit()` instead for integers?

Reply:
yeah digit seems better here, updated